kuksa_databroker/databroker/src/broker.rs

```diff
@@ -868,6 +877,28 @@ impl<'a> Iterator for EntryIterator<'a> {
     }
 }
 
+impl<'a, 'b> EntryReadAccess<'a, 'b> {
+    pub fn datapoint(&self) -> Result<&'a Datapoint, ReadError> {
+        match self.permissions.can_read(&self.entry.metadata.path) {
+            Ok(_) => Ok(&self.entry.datapoint),
+            Err(PermissionError::Denied) => Err(ReadError::PermissionDenied),
+            Err(PermissionError::Expired) => Err(ReadError::PermissionExpired),
+        }
+    }
+
+    pub fn actuator_target(&self) -> Result<&Option<Datapoint>, ReadError> {
+        match self.permissions.can_read(&self.entry.metadata.path) {
+            Ok(_) => Ok(&self.entry.actuator_target),
+            Err(PermissionError::Denied) => Err(ReadError::PermissionDenied),
+            Err(PermissionError::Expired) => Err(ReadError::PermissionExpired),
+        }
+    }
+
+    pub fn metadata(&self) -> &Metadata {
+        &self.entry.metadata
+    }
```

Comment:
General I saw it in DatabrokerReadAccess neither, but don't we state in https://github.com/eclipse/kuksa.val/blob/master/doc/KUKSA.val_data_broker/authorization.md that 
```
read:Vehicle.Speed | Allow client to read Vehicle.Speed (including metadata)
-- | --
``` 
metadata needs read access too?

Reply:
That's true.
I have a feeling metadata should be readable by all. But I could be wrong.
It's not enforced in the other accessors either, so if we decide that you need `read` access in order to read metadata, those would need to change as well.

Reply:
Yeah, I see that it's more useful if everyone can access it. Just had a look at documentation and I think we should clarify it there at least.

Reply:
In the long run I think we might get requests to restrict the possibility to read metadata. Not useful if the server/broker only has a standard VSS file as config, but if it contains a lot of private (OEM-specific) signals then just showing their names (and descriptions) could reveal sensitive information on how certain problems have been solved in that vehicle, One can compare with DBC or ARXML files which typically are considered confidential by OEMs. 

But for now I think it would be sufficient if the limitation is mentioned in documentation. Maybe we should also have a concept on how the token format needs to be changed/extended to allow managing metadata access.

Reply:
I'm actually starting to think it might be better to change databroker according to what the documentation says (as pointed out by @lukasmittag).

You get access to the metadata (implicitly) as soon as you have some form of access to that entry. Otherwise you don't.

And an extension of that could be that based on _what_ form of access you have `read`, `actuate` / `provide`, the metadata you would have access to would be different?